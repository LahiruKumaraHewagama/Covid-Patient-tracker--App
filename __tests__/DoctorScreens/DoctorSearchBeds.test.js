import React from "react";
import DoctorSearchBeds from "../../screens/DoctorScreens/DoctorSearchBeds";
import { act, create } from "react-test-renderer";
import { render, fireEvent, waitFor, cleanup } from '@testing-library/react-native';

jest.useFakeTimers();
const navigation = { addListener: jest.fn() }

afterEach(cleanup);

let tree
test('snapshot Doctor Search Beds Results page', async () => {  
    act(() => { tree = create(<DoctorSearchBeds navigation={navigation} />) })   
        expect(tree).toMatchSnapshot();
    })

test('Facility empty input error massage', async () => {
    global.alert = jest.fn();
    const searchBtn = tree.root.findByProps({ testID: 'search' }).props;
    act(() => searchBtn.onPress());
    expect(global.alert).toHaveBeenCalledWith("Please select Facility !");
});

test('Test Result empty input error massage', async () => {
    global.alert = jest.fn();
    const  facility = tree.root.findByProps({ testID: 'facility_test' }).props;
    act(() => facility.onValueChange("National Hospital of Sri Lanka"));
    const searchBtn = tree.root.findByProps({ testID: 'search' }).props;
    act(() => searchBtn.onPress());
    act(() => expect(global.alert.length).toEqual(0));
    

});

