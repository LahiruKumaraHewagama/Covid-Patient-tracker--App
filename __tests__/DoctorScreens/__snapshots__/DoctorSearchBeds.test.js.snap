// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot Doctor Search Beds Results page 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 3,
      "paddingHorizontal": 20,
      "paddingVertical": 20,
    }
  }
>
  <View
    style={
      Object {
        "justifyContent": "flex-end",
        "paddingBottom": 20,
        "paddingHorizontal": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#009387",
          "fontSize": 30,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      SEARCH  BEDS
    </Text>
    <View
      style={
        Object {
          "borderBottomColor": "#009387",
          "borderBottomWidth": 2,
        }
      }
    />
  </View>
  <RCTScrollView
    style={
      Object {
        "marginTop": 50,
        "paddingRight": 20,
      }
    }
  >
    <View>
      <Text
        style={
          Object {
            "color": "#007c7a",
            "fontSize": 16,
          }
        }
      >
        Facility Name
      </Text>
      <View
        style={
          Array [
            Object {
              "flex": 0.5,
            },
            Object {
              "alignSelf": "center",
              "borderColor": "#007c7a",
              "borderRadius": 10,
              "borderWidth": 1,
              "marginLeft": 5,
              "marginRight": 5,
              "marginTop": 10,
              "width": 300,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "borderColor": "#007c7a",
              "borderRadius": 20,
              "borderWidth": 1,
              "flexDirection": "row",
              "marginTop": 5,
              "padding": 5,
            }
          }
        >
          <RNCPicker
            items={
              Array [
                Object {
                  "label": "Select",
                  "testID": undefined,
                  "textColor": 4289374890,
                  "value": "disabled",
                },
              ]
            }
            onChange={[Function]}
            selectedIndex={0}
            style={
              Array [
                Object {
                  "height": 216,
                },
                undefined,
              ]
            }
            testID="facility_test"
          />
        </View>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#009387",
            "borderColor": "#20d1ce",
            "borderRadius": 10,
            "borderWidth": 2,
            "marginBottom": 20,
            "marginLeft": 20,
            "marginRight": 20,
            "marginTop": 20,
            "opacity": 1,
            "paddingHorizontal": 14,
            "paddingVertical": 10,
          }
        }
        testID="search"
      >
        <Text
          style={
            Object {
              "color": "#fff",
              "fontSize": 20,
            }
          }
        >
          Search
        </Text>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
